litestar/middleware/compression/facade.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, ClassVar, Protocol
+
+if TYPE_CHECKING:
+    from io import BytesIO
+
+    from litestar.config.compression import CompressionConfig
+    from litestar.enums import CompressionEncoding
+
+
+class CompressionFacade(Protocol):
+    """A unified facade offering a uniform interface for different compression libraries."""
+
+    encoding: ClassVar[str]
+
+    def __init__(
+        self, buffer: BytesIO, compression_encoding: CompressionEncoding | str, config: CompressionConfig
+    ) -> None:
+        """Initialize ``CompressionFacade``.
+
+        Args:
+            buffer: A bytes IO buffer to write the compressed data into.
+            compression_encoding: The compression encoding used.
```

Comment:
Once again, we don't actually need the `compression_encoding`, but it's being kept for compatibility.